timemachine/fe/single_topology_v3.py

```diff
@@ -269,16 +270,32 @@ def setup_end_state(ff, mol_a, mol_b, core, a_to_c, b_to_c):
     mol_c_torsion_idxs = mol_c_proper_idxs + mol_c_improper_idxs
     mol_c_torsion_params = mol_c_proper_params + mol_c_improper_params
 
-    bond_potential = potentials.HarmonicBond(mol_c_bond_idxs).bind(mol_c_bond_params)
-    angle_potential = potentials.HarmonicAngle(mol_c_angle_idxs).bind(mol_c_angle_params)
-    torsion_potential = potentials.PeriodicTorsion(mol_c_torsion_idxs).bind(mol_c_torsion_params)
+    # canonicalize bonds
+    mol_c_bond_idxs_canon = [canonicalize_bond(idxs) for idxs in mol_c_bond_idxs]
+    bond_potential = potentials.HarmonicBond(mol_c_bond_idxs_canon).bind(mol_c_bond_params)
+
+    mol_c_angle_idxs_canon = [canonicalize_bond(idxs) for idxs in mol_c_angle_idxs]
+    angle_potential = potentials.HarmonicAngle(mol_c_angle_idxs_canon).bind(mol_c_angle_params)
+
+    mol_c_torsion_idxs_canon = [canonicalize_bond(idxs) for idxs in mol_c_torsion_idxs]
+    torsion_potential = potentials.PeriodicTorsion(mol_c_torsion_idxs_canon).bind(mol_c_torsion_params)
```

Comment:
Point of uncertainty: Should proper and improper torsion idxs be canonicalized differently?

(Consider if mol_a has an improper (0, 1, 2, 3), and mol_b has an improper (3, 1, 2, 0) -- `canonicalize_bond` will map these to (0, 1, 2, 3) and (0, 2, 1, 3), i.e. not be able to recognize these as describing the same set of neighbors {0, 2, 3} around the same central atom 1...)

Reply:
^ Based on offline discussion with @proteneer -- there is a fix to apply here, but it's more involved than I was originally thinking. Also note the example I provided contains an error (assumed central atom is in tuple[1], but it should be tuple[0]).